### PR TITLE
feat(api-v1): Change API v1 file uploads to work like API v2

### LINKED ISSUE #535: Don't require Knora and Sipi to share a directory for image upload
At the moment, the file upload through the ``webapi`` requires that there is a shared directory with ``sipi``. This limits the way how the ``webapi`` and ``sipi`` can be deployed. Either they need to be running on the same server or both have access to the same network share.

### PR DESCRIPTION
**Pull request by benjamingeer**
This PR changes API v1 so that Knora and Sipi no longer need to share a filesystem (#535).

- [x] Change API v1 to use the same Sipi scripts as API v2.
- [x] Remove the API v1 Sipi scripts.
- [x] Change `salsah1` to use `upload.lua`, sending it a JWT token.
- [ ] Once #1532 is done, fix `KnoraSipiIntegrationV1ITSpec` to use API v2 to create the `XSLTransformation` resource.
- [x] Update other API v1 tests.
- [x] Update API v1 docs.
- [x] Update release notes.

Needs #1532.
Resolves #535.

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@benjamingeer I think for this PR the Python load test script has to be adapted. 

Could you please instruct me how I have to adapt my import scripts for the v1 non GUI case?

**Comment by tobiasschweizer** [comment]
and then the members would not be static:

```
@Injectable()
export class AppInitService {

    settings: IAppConfig;
    coreConfig: KuiCoreConfig;

    constructor() {
    }

    Init() {

        return new Promise<void>((resolve, reject) => {
            // console.log('AppInitService.init() called');
            // do your initialisation stuff here

            const data = <IAppConfig> window['tempConfigStorage'];
            // console.log('AppInitService: json', data);
            this.settings = data;

            this.coreConfig = <KuiCoreConfig> {
                name: this.settings.appName,
                api: this.settings.apiURL,
                media: this.settings.iiifURL,
                app: this.settings.appURL
            };

            // console.log('AppInitService: finished');

            resolve();
        });
    }
}
```

**Comment by benjamingeer** [comment]
> Could you please instruct me how I have to adapt my import scripts for the v1 non GUI case?

You’ll have to use v2. When this PR is done, there will be documentation. But I still have a lot of work left to do on it.

**Comment by tobiasschweizer** [comment]
Ok, can we coodinate that next week? Also with @SepidehAlassi. This will affect the BEOL import process. Also I won‘t work in the week of carnival.

**Comment by tobiasschweizer** [comment]
@benjamingeer Could I start using v2 next week? Then I could already adapt my import scripts (image file values, XSL transformations stored in Sipi).

**Comment by benjamingeer** [comment]
@tobiasschweizer This PR may take some time because it depends on PR #1206, which depends on work that Lukas is doing in Sipi. But you can switch to v2 now. The documentation on creating file values in v2 is here:

https://docs.knora.org/paradox/03-apis/api-v2/editing-values.html#creating-file-values

**Comment by tobiasschweizer** [comment]
@benjamingeer Ok, I will have a look at it next week. Thanks.

**Comment by benjamingeer** [comment]
@tobiasschweizer I'm trying to figure out what to do about the `/v1/mapping` route, which accepted a multipart request. I'm confused because [the API v2 docs describe a /v2/mapping route](https://docs.knora.org/paradox/03-apis/api-v2/xml-to-standoff-mapping.html#validating-a-mapping-and-sending-it-to-knora), but that route doesn't seem to exist. Is this a documentation bug? If so, once PR #1206 is done, I would like to get rid of the `/v1/mapping` route and make a new `/v2/mapping` route that accepts the filename of an XML file that has already been uploaded to Sipi. I might need your help with that.

**Comment by benjamingeer** [comment]
@tobiasschweizer OK, now I'm not so confused, I found the `/v2/mapping` route in `StandoffRouteV2`, and I realised that the `/v1/mapping` and `/v2/mapping` routes don't use Sipi. It's only the XSLT transformation that's stored in Sipi. So we just need PR #1206 so we can upload the XSLT file to Sipi the v2 way.

**Comment by benjamingeer** [comment]
TODO: once #1206 is done, change `/v2/mapping` to use an XML file already uploaded to Sipi. Consider getting rid of `/v1/mapping`.

**Comment by tobiasschweizer** [comment]
Yes, the IRI of an XSL transformation can be referenced from a mapping as a default transformation. The XSLT file is stored in Sipi when creating the representation in Knora (v1).

**Comment by benjamingeer** [comment]
@tobiasschweizer It would be great if you could review this sometime in the next week. I can't finish it until #1206 is done, but in the meantime, you could try changing your import script to use new API v1 image upload procedure (no need to switch to v2 if you don't want to). I've updated the API v1 docs to explain how this works now.

**Comment by tobiasschweizer** [comment]
@benjamingeer I wanted to look at this PR this week but now I realize that I won't manage. I will look at it it when I am back to the office on Monday, March 18 (after carnival).

File: docs/src/paradox/00-release-notes/next.md [hunk 1]
```diff
@@ -15,3 +15,12 @@ Also, please change the **HINT** to the appropriate level:
 ## HINT => MAJOR CHANGE
 
 - FIX: Unescape standoff string attributes when verifying text value update (@github[#1242](#1242))
+
+- MAJOR: Change API v1 file uploads to work like API v2 (@github[#1233](#1233)). To enable
+  Knora and Sipi tow work without sharing a filesystem, the procedure
+  for uploading files in API v1 has changed; see
```
Comment (gfoo) [inline docs/src/paradox/00-release-notes/next.md]: `to work`  typo master has said :)

**Comment by tobiasschweizer** [comment]
@benjamingeer I can confirm that the new API v1 image upload procedure works.

Once it also supports XML files, I will try that too.

**Comment by subotic** [comment]
This pull request has been mentioned on **Discuss DaSCH**. There might be relevant details there:

https://discuss.dasch.swiss/t/support-non-image-files-in-knora-api-v2/33/1

**Comment by tobiasschweizer** [comment]
@benjamingeer once this PR is ready, could you let me test it with my adapted import scripts?

**Comment by benjamingeer** [comment]
@tobiasschweizer Will ask you to review it when it’s ready. ;)

**Comment by tobiasschweizer** [comment]
> Will ask you to review it when it’s ready. ;-)

That makes sense :-)

**Comment by subotic** [comment]
@benjamingeer I'm preparing a [PR](https://github.com/dasch-swiss/sipi/pull/324) which should contain the necessary fix. If everything works out. I will publish a release candidate of SIPI today.

**Comment by benjamingeer** [comment]
@subotic Great news, thanks!

**Comment by subotic** [comment]
@benjamingeer I've bumped the SIpi version in this branch to the latest RC. Let me know if you need help.

**Comment by subotic** [comment]
@benjamingeer I've missed something when I changed the Sipi version. I will let you know when its done.

**Comment by benjamingeer** [comment]
@subotic Before I finish this PR, I need to do a new PR for https://dasch.myjetbrains.com/youtrack/issue/DSP-44.

**Pull request merged**